addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/service/SmappeeService.java

```diff
@@ -0,0 +1,294 @@
+    private final Gson gson;
+
+    private String accessToken;
+    private DateTime accessTokenValidity;
+    private String refreshToken;
+
+    public SmappeeService(String client_id, String client_secret, String username, String password,
+            String serviceLocationName, long refreshInterval) {
+        this.serviceLocationName = serviceLocationName;
+        this.client_id = client_id;
+        this.client_secret = client_secret;
+        this.username = username;
+        this.password = password;
+
+        this.retry = 1;
+        this.refreshInterval = refreshInterval;
+
+        this.gson = new Gson();
+
+        this.isInitialized = false;
+    }
+
+    public void startAutomaticRefresh(ScheduledExecutorService scheduledExecutorService,
+            final ReadingsUpdate readingsUpdate) {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    SmappeeDeviceReading readings = getDeviceReadings();
+                    if (readings.consumptions.length > 0) {
+                        readingsUpdate.newState(readings);
+                    }
+                } catch (InvalidConfigurationException e) {
+                    readingsUpdate.invalidConfig();
+                }
+            }
+        };
+
+        scheduledJob = scheduledExecutorService.scheduleAtFixedRate(runnable, 0, this.refreshInterval,
+                TimeUnit.MILLISECONDS);
+    }
+
+    ScheduledFuture<?> scheduledJob;
+
+    public void stopAutomaticRefresh() {
+        scheduledJob.cancel(true);
+    }
+
+    public SmappeeDeviceReading getDeviceReadings() throws InvalidConfigurationException {
+        int currentTry = 0;
+        do {
+            try {
+                DateTime nowUtc = DateTime.now(DateTimeZone.UTC).minusMinutes(1);
+                DateTime nowUtcMinus20Min = DateTime.now(DateTimeZone.UTC).minusMinutes(20);
+
+                String nowUtcMillis = String.valueOf(nowUtc.getMillis());
+                String nowUtcMinus20MinMillis = String.valueOf(nowUtcMinus20Min.getMillis());
+
+                // https://app1pub.smappee.net/dev/v1/servicelocation/123/consumption?aggregation=1&from=1388534400000&to=1391212800000
+
+                String responseReadings = GetData("/dev/v1/servicelocation/" + this.serviceLocationId
+                        + "/consumption?aggregation=1&from=" + nowUtcMinus20MinMillis + "&to=" + nowUtcMillis);
+
+                SmappeeDeviceReading readings = gson.fromJson(responseReadings, SmappeeDeviceReading.class);
+
+                // Sum of 5 minutes in Wh so average power is 60/5 x
+
+                for (SmappeeDeviceReadingConsumption consumption : readings.consumptions) {
+                    consumption.consumption = consumption.consumption * 12;
+                    consumption.solar = consumption.solar * 12;
+                }
+
+                logger.debug("smappee'{}' read", new Object[] { this.serviceLocationId });
```

Comment:
no need to do new Object, just pass the value directly as argument.

Reply:
ok